py/picca/delta_extraction/expected_fluxes/true_continuum.py

```diff
@@ -396,65 +397,57 @@ def read_raw_statistics(self):
-                    f"raw\t{header['L_MIN']}\t{header['L_MAX']}\t"
-                    f"{header['LR_MIN']}\t{header['LR_MAX']}\t{header['DEL_LL']}"
-                    f"input\t{log_lambda_min}\t{log_lambda_max}\t"
-                    f"{log_lambda_rest_min}\t{log_lambda_rest_max}"
-                    "provide a custom file in 'raw statistics file' field "
-                    "matching input pixelization scheme")
-            log_lambda = hdul[1]['LAMBDA'][:]
+            pixel_step_key = 'DEL_LL'
+            log_lambda = fits_data['LAMBDA']
+
+            if header['LINEAR']:
+                is_rawfile_consistent = False
+                err_msg += "File wave solution is linear.\n"
+
         elif Forest.wave_solution == "lin":
-            pixel_step = 10**Forest.log_lambda_grid[
-                1] - 10**Forest.log_lambda_grid[0]
-            lambda_min = 10**Forest.log_lambda_grid[0]
-            lambda_max = 10**Forest.log_lambda_grid[-1] - pixel_step / 2
-            lambda_rest_min = 10**Forest.log_lambda_rest_frame_grid[
-                0] - pixel_step / 2
-            lambda_rest_max = 10**Forest.log_lambda_rest_frame_grid[-1]
-            if (not header['LINEAR'] or
-                    not np.isclose(header['L_MIN'], lambda_min, rtol=1e-3) or
-                    not np.isclose(header['L_MAX'], lambda_max, rtol=1e-3) or
-                    not np.isclose(header['LR_MIN'], lambda_rest_min, rtol=1e-3)
-                    or
-                    not np.isclose(header['LR_MAX'], lambda_rest_max, rtol=1e-3)
-                    or not np.isclose(header['DEL_L'],
-                                      10**Forest.log_lambda_grid[1] -
-                                      10**Forest.log_lambda_grid[0],
-                                      rtol=1e-3)):
-                raise ExpectedFluxError(
-                    "raw statistics file pixelization scheme does not match "
-                    "input pixelization scheme. "
-                    "\t\tL_MIN\tL_MAX\tLR_MIN\tLR_MAX\tDEL_L"
-                    f"raw\t{header['L_MIN']}\t{header['L_MAX']}\t"
-                    f"{header['LR_MIN']}\t{header['LR_MAX']}\t{header['DEL_L']}"
-                    f"input\t{lambda_min}\t{lambda_max}\t{lambda_rest_min}\t"
-                    f"{lambda_rest_max} provide a custom file in 'raw "
-                    "statistics file' field matching input pixelization scheme")
-            log_lambda = np.log10(hdul[1]['LAMBDA'][:])
-
-        flux_variance = hdul[1]['VAR'][:]
-        mean_flux = hdul[1]['MEANFLUX'][:]
-        hdul.close()
+            pixel_step = 10**Forest.log_lambda_grid[1] - 10**Forest.log_lambda_grid[0]
+            pixel_step_key = 'DEL_L'
+            log_lambda = np.log10(fits_data['LAMBDA'])
+
+            if not header['LINEAR']:
+                is_rawfile_consistent = False
+                err_msg += "File wave solution is not linear.\n"
+
+        def _check_header_consistency(_key, _test_val, atol):
```

Comment:
This function is the source of the pylint errors.
The reason is that you're using variables that are not defined within the scope of the function (err_msg) or are potentially just defined in the scope of the function (is_rawfile_consistent), i.e. cannot be accessed outside.
For this to work the variables would either need to be declared as global (not sure if that works and also bad style) or would need to be passed to the function by reference (I'm never sure how to guarantee this in python for stuff like a single bool) or by value and then returned in the end...

Reply:
I thought locally declared functions can access variables outside. We shouldn't declare global. I will rewrite to return values, but this just looked cleaner.